src/VisualStudio/Core/Def/Implementation/Library/ObjectBrowser/AbstractObjectBrowserLibraryManager.cs

```diff
@@ -94,10 +94,7 @@ private void OnWorkspaceChanged(object sender, WorkspaceChangeEventArgs e)
                     var oldDocument = e.OldSolution.GetDocument(e.DocumentId);
                     var newDocument = e.NewSolution.GetDocument(e.DocumentId);
 
-                    // make sure we do this in background thread. we don't care about ordering of events
-                    // we just need to refresh OB at some point if it ever needs to be updated
-                    // link to the bug tracking root cause  - https://devdiv.visualstudio.com/DefaultCollection/DevDiv/_workitems?id=169649&_a=edit
-                    Task.Run(() => DocumentChangedAsync(oldDocument, newDocument));
+                    UpdateDocument(oldDocument, newDocument);
```

Comment:
this does worry me, as it will do all this processing synchronously.

Reply:
nevermind.  all this does is update version numbers, which then end up causing updates down the line.

Reply:
In the past, it ran separately because it had actual work to do. Now it doesn't do that work so it can run synchronously again.

Reply:
agreed.  this part seems fine.  we may overl agressive bump the version numbers here.  but i'm ok with that on a branch switch as all it means is that VS will eventually refresh this stuff.